src/web/mjs/connectors/kakaopage.mjs

```diff
@@ -28,52 +28,459 @@ export default class kakaopage extends Connector {
     }
 
     async _getChapters(manga) {
-        let chapterList = [];
-        for(let page = 0, run = true; run; page++) {
-            let chapters = await this._getChaptersFromPage(manga, page);
-            chapters.length > 0 ? chapterList.push(...chapters) : run = false;
+        let nextCursor = '';
+        const chapterList = [];
+        for (let run = true; run;) {
+            const data = await this._getChaptersFromPage(manga, nextCursor);
+
+            const chapters = data.contentHomeProductList.edges.map(chapter => {
+                return {
+                    id : chapter.node.single.productId,
+                    title : chapter.node.single.title.replace(manga.title, '').trim()
+                };
+            });
+
+            nextCursor = data.contentHomeProductList.pageInfo.hasNextPage ? data.contentHomeProductList.pageInfo.endCursor : null;
+            chapterList.push(...chapters);
+            if (nextCursor == null) {
+                run = false;
+            }
         }
         return chapterList;
     }
 
-    async _getChaptersFromPage(manga, page) {
-        let form = new FormData();
-        form.append('seriesid', manga.id);
-        form.append('page', page);
-        form.append('direction', 'desc');
-        form.append('page_size', 20);
-        form.append('without_hidden', true);
-        let request = new Request('https://api2-page.kakao.com/api/v5/store/singles', {
-            ...this.requestOptions,
-            method: 'POST',
-            body: form
-        });
-        let data = await this.fetchJSON(request);
-        return data.singles.map(ele => {
-            return{
-                id: ele.id,
-                title: ele.title.trim()
-            };
-        });
+    async _getChaptersFromPage(manga, nextCursor) {
+        const gql = {
+            operationName: 'contentHomeProductList',
+            variables: {
+                boughtOnly: false,
+                seriesId : manga.id,
+                sortType : 'asc',
+                after : nextCursor,
+            },
+            query: `query contentHomeProductList($after: String, $before: String, $first: Int, $last: Int, $seriesId: Long!, $boughtOnly: Boolean, $sortType: String) {
```

Comment:
GQL query can probably be simplified, I'm pretty sure the connector only needs a very small amount of all those fields

Reply:
I tried of but its fragments of fragments of fragments :D. Well, i'll see what i can do